impl/shim/jaas/src/main/java/org/pentaho/big/data/impl/shim/jaas/JaasConfigServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Big Data
+ *
+ * Copyright (C) 2002-2017 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.big.data.impl.shim.jaas;
+
+import org.pentaho.bigdata.api.jaas.JaasConfigService;
+
+import java.util.Properties;
+
+public class JaasConfigServiceImpl implements JaasConfigService {
+  public static final String KERBEROS_PRINCIPAL = "pentaho.authentication.default.kerberos.principal";
+  public static final String KERBEROS_KEYTAB = "pentaho.authentication.default.kerberos.keytabLocation";
+  private Properties configProperties;
+
+  public JaasConfigServiceImpl( Properties configProperties ) {
+
+    this.configProperties = configProperties;
+  }
+
+  @Override public String getJaasConfig() {
+    return
+      "com.sun.security.auth.module.Krb5LoginModule required\n"
+        + "useKeyTab=true\n"
+        + "serviceName=kafka\n"
```

Comment:
serviceName is the name of the service configured at broker side. Default value for CDH and HDP is "kafka", but it might be changed by cluster admin. It maybe makes sense to allow specifying this in config.properties and use "kafka" as default value.